azure-kit-starter/src/main/java/com/vaadin/azure/starter/AzureKitConfiguration.java

```diff
@@ -0,0 +1,28 @@
+package com.vaadin.azure.starter;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.session.SaveMode;
+import org.springframework.session.config.SessionRepositoryCustomizer;
+import org.springframework.session.data.redis.RedisIndexedSessionRepository;
+
+/**
+ * This configuration bean is provided to auto-configure Vaadin apps to run in a
+ * clustered environment.
+ */
+@AutoConfiguration
+public class AzureKitConfiguration {
+
+    /**
+     * Provides a {@link SessionRepositoryCustomizer} bean to configure
+     * {@link RedisIndexedSessionRepository} to work with Vaadin sessions.
+     *
+     * @return the session-repository customizer
+     */
+    @Bean
```

Comment:
Shall we add a `@ConditionalOnBean(RedisIndexedSessionRepository.class)` annotation?

Reply:
We can't use `@ConditionalOnBean` since the bean is defined after being customized, so we need the customizer bean available before that. But I made it `@ConditionalOnClass` and marked Redis dependencies as optional, so using the starter does not transitively add them!